File: rust/src/shell_pilot.rs
use std::time::Duration;
use tokio::time::sleep;

pub struct ShellPilotAgent {}

impl ShellPilotAgent {
    pub fn new() -> Self {
        Self {}
    }

    pub async fn predict_next_command(&self, history: Vec<String>) -> Result<String, String> {
        sleep(Duration::from_millis(150)).await;
        Ok(format!("Predicted command based on {} history items: git status", history.len()))
    }
}

File: go/internal/agent/shell_pilot.go
package agent

import (
	"context"
	"fmt"
	"time"
)

type ShellPilotAgent struct {
	BaseAgent
}

func NewShellPilotAgent() *ShellPilotAgent {
	return &ShellPilotAgent{}
}

func (a *ShellPilotAgent) PredictNextCommand(ctx context.Context, history []string) (string, error) {
	time.Sleep(150 * time.Millisecond)
	return fmt.Sprintf("Predicted command based on %d history items: git status", len(history)), nil
}
